File: src/main/kotlin/com/ks/culinario/network/dto/ShoppingListDTO.kt
package com.ks.culinario.network.dto

data class ShoppingListDTO(
    val id: Int,
    val title: String,
    val description: String,
    val items: List<ShoppingListItemDTO>
)

data class NewShoppingListDTO(
    val title: String,
    val description: String,
    val items: List<ShoppingListItemDTO>
)

data class ShoppingListItemDTO(
    val name: String,
    val amount: Int
)

File: dotnet/Culinario.Application/DTOs/ShoppingListDTO.cs
namespace Culinario.Application.DTOs;

public record ShoppingListDTO(
    int Id,
    string Title,
    string Description,
    List<ShoppingListItemDTO> Items
);

public record NewShoppingListDTO(
    string Title,
    string Description,
    List<ShoppingListItemDTO> Items
)
{

    public NewShoppingListDTO() : this(string.Empty, string.Empty, new List<ShoppingListItemDTO>()) { }
}

public record ShoppingListItemDTO(string Name, int Amount)
{
    public ShoppingListItemDTO() : this(string.Empty, 0) { }
}
